x/stakeibc/keeper/update_validator_shares_exch_rate_test.go

```diff
@@ -0,0 +1,236 @@
+
+type QueryValidatorExchangeRateTestCase struct {
+	msg                types.MsgUpdateValidatorSharesExchRate
+	currentEpoch       uint64
+	hostZone           types.HostZone
+	strideEpochTracker types.EpochTracker
+	dayEpochTracker    types.EpochTracker
+}
+
+func (s *KeeperTestSuite) SetupQueryValidatorExchangeRate() QueryValidatorExchangeRateTestCase {
+	currentEpoch := uint64(1)
+	valoperAddr := "cosmosvaloper133lfs9gcpxqj6er3kx605e3v9lqp2pg5syhvsz"
+
+	// set up IBC
+	s.CreateTransferChannel(HostChainId)
+
+	hostZone := types.HostZone{
+		ChainId:      HostChainId,
+		HostDenom:    Atom,
+		IBCDenom:     IbcAtom,
+		Bech32Prefix: Bech32Prefix,
+	}
+
+	s.App.StakeibcKeeper.SetHostZone(s.Ctx(), hostZone)
+
+	// This will make the current time 90% through the epoch
+	strideEpochTracker := types.EpochTracker{
+		EpochIdentifier:    epochtypes.STRIDE_EPOCH,
+		EpochNumber:        currentEpoch,
+		Duration:           10_000_000_000,                                               // 10 second epochs
+		NextEpochStartTime: uint64(s.Coordinator.CurrentTime.UnixNano() + 1_000_000_000), // epoch ends in 1 second
+	}
+	s.App.StakeibcKeeper.SetEpochTracker(s.Ctx(), strideEpochTracker)
+
+	// This will make the current time 50% through the day
+	dayEpochTracker := types.EpochTracker{
+		EpochIdentifier:    epochtypes.DAY_EPOCH,
+		EpochNumber:        currentEpoch,
+		Duration:           40_000_000_000,                                                // 40 second epochs
+		NextEpochStartTime: uint64(s.Coordinator.CurrentTime.UnixNano() + 20_000_000_000), // day ends in 20 second
+	}
+	s.App.StakeibcKeeper.SetEpochTracker(s.Ctx(), dayEpochTracker)
+
+	return QueryValidatorExchangeRateTestCase{
+		msg: types.MsgUpdateValidatorSharesExchRate{
+			Creator: s.TestAccs[0].String(),
+			ChainId: HostChainId,
+			Valoper: valoperAddr,
+		},
+		currentEpoch:       currentEpoch,
+		hostZone:           hostZone,
+		strideEpochTracker: strideEpochTracker,
+		dayEpochTracker:    dayEpochTracker,
+	}
+}
+
+func (s *KeeperTestSuite) TestQueryValidatorExchangeRate_Successful() {
+	tc := s.SetupQueryValidatorExchangeRate()
+
+	resp, err := s.App.StakeibcKeeper.QueryValidatorExchangeRate(s.Ctx(), &tc.msg)
+	s.Require().NoError(err, "no error expected")
+	s.Require().NotNil(resp, "response should not be nil")
+
+	// check a query was created (a simple test; details about queries are covered in makeRequest's test)
+	queries := s.App.InterchainqueryKeeper.AllQueries(s.Ctx())
+	s.Require().Len(queries, 1, "one query should have been created")
+}
+
+func (s *KeeperTestSuite) TestQueryValidatorExchangeRate_BeforeBufferWindow() {
+	tc := s.SetupQueryValidatorExchangeRate()
+
+	// set the time to be 50% through the stride_epoch
+	strideEpochTracker := tc.strideEpochTracker
+	strideEpochTracker.NextEpochStartTime = uint64(s.Coordinator.CurrentTime.UnixNano() + int64(strideEpochTracker.Duration)/2) // 50% through the epoch
```

Comment:
nit: rm `// 50% through the epoch` (since its already stated at the top)

Reply:
done